frontend/catalyst/debug/compiler_functions.py

```diff
@@ -160,3 +161,43 @@ def compile_from_mlir(ir, compiler=None, compile_options=None):
         result_types = [mlir.ir.RankedTensorType.parse(rt) for rt in func_data[1].split(",")]
 
     return CompiledFunction(shared_object, qfunc_name, result_types, None, compiler.options)
+
+
+@debug_logger
+def compile_cmain(fn, *args):
+    """Generate and compile a C program that calls a jitted function with the provided arguments.
+
+    Args:
+        fn (QJIT): a qjit-decorated function
+        *args: argument values to use in the C program when invoking ``fn``
+
+    Returns:
+        (str): the paths that should be included in LD_LIBRARY_PATH.
+        (str): the path of output binary.
+    """
+    f_name = str(fn.__name__)
+    workspace = str(fn.workspace)
+    main_c_file = workspace + "/main.c"
+    output_file = workspace + "/" + f_name + ".out"
+    shared_object_file = workspace + "/" + f_name + ".so"
+    lib_shared_object_file = workspace + "/lib" + f_name + ".so"
+    options = fn.compiler.options
+    with open(main_c_file, "w", encoding="utf-8") as file:
+        file.write(get_cmain(fn, *args))
+    shutil.copy(shared_object_file, lib_shared_object_file)
+
+    # configure flags
+    default_flags = LinkerDriver.get_default_flags(options)
+    no_shared_flags = [fs for fs in default_flags if fs != "-shared"]
+    link_so_flags = no_shared_flags + [
+        "-Wl,-rpath," + workspace,
+        "-L" + workspace,
+        "-l" + f_name,
+        "-g",
```

Comment:
Not sure how useful `-g` is since it only helps the main function, but since the intention for now is that this is a debug feature, it can't hurt either.